etc/ncp-config.d/nc-previews-auto.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+# Periodically generate previews for the gallery
+#
+# Copyleft 2018 by Timo Stiefel and Ignacio Nunez Hernanz <nacho _a_t_ ownyourbits _d_o_t_ com>
+# GPL licensed (see end of file) * Use at your own risk!
+#
+
+ACTIVE_=no
+STARTTIME_=240
+RUNTIME_=180
+SMALLONLY_=yes
+
+DESCRIPTION="Periodically generate previews for the gallery"
+INFO="Set the STARTTIME in minutes after midnight and RUNTIME in minutes.
+
+Make sure that you have the correct time zone set.
+You can use 'sudo tzselect' in shell for that.
+
+Activate SMALLONLY for preventing the generation of big preview files that are seldom used.
+They're still big enough for gallery previews."
+
+configure()
+{
+  [[ $ACTIVE_ != "yes" ]] && { 
+    rm /etc/cron.d/nc-previews-auto
+    service cron restart
+    echo "Automatic preview generation disabled"
+    return 0
+  }
+  
+  # Required values for SMALLONLY previews (in pixel as used with standard DPI screens without zooming):
+  #   Thumbnail in files list: square 32
+  #   Sidebar preview (when opening "Details"): width 512; (also height 256?)
+  #   Gallery folder preview: width 128, 256; heigth 128, 256
+  #   Gallery preview: width 512; height 256
+  # 
+  if [ $SMALLONLY_ == "yes" ]
+    then
+      sudo -u www-data php /var/www/nextcloud/occ config:app:set --value="32"  previewgenerator squareSizes
```

Comment:
looks great! good job

Reply:
I had been thinking about adding

```
'preview_max_x' => 1024,
'preview_max_y' => 768,
'preview_max_scale_factor' => 1,n
```

to the previews app. I think that we should add this and your options either as the default choice, or even just hardcoded for NCPi users

Reply:
I wouldn't hardcode `preview_max_x` and  `preview_max_y` as they also affect the presentation of images in the gallery (not only the overview).
I personally would also not include it in SMALLONLY, for that bigger previews (eg. for the gallery) are still generated if really needed.

How about adding another option with the recommendation to enable it in lowest-end devices? Or do we then have too many optional stuff?

Reply:
I could live with hardcoding the SMALLONLY functionality, but wouldn't do it for `preview_max`.

Reply:
Well, I think it is better to keep things intuitive, I try to avoid options if possible but that's me